.github/workflows/master.yml

```diff
@@ -79,7 +79,7 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           changed_files=$(python dev/list_changed_files.py --repository mlflow/mlflow --pr-num ${{ github.event.number }})
-          if [ ! -z "$(echo $changed_files | grep '\.py$')"]; then
+          if echo "$changed_files" | grep -q '\.py'; then
```

Comment:
https://github.com/mlflow/mlflow/actions/runs/5253971877/jobs/9491992419?pr=8715 is a failed build. Took me a while to understand what went wrong.

```
+ '[' '!' -z 'tests/store/artifact/test_mlflow_artifact_repo.py]'
/home/runner/work/_temp/03e1cc30-2eb0-45ac-b021-ce8bb2ed845a.sh: line 2: [: missing `]'
```

`[ ! -z "$(echo $changed_files | grep '\.py$')"]` fails and exits with 1. 1 means `False` for `if`. There is no `else`. Nothing is executed.